src/lightning/fabric/fabric.py

```diff
@@ -632,24 +643,17 @@ def no_backward_sync(self, module: _FabricModule, enabled: bool = True) -> Gener
                 " `model = fabric.setup(model, ...)`"
             )
         if not enabled or isinstance(self._strategy, (SingleDeviceStrategy, XLAStrategy)):
-            context = nullcontext()
-        elif self._strategy._backward_sync_control is None:
+            return nullcontext()
+        if self._strategy._backward_sync_control is None:
             rank_zero_warn(
                 f"The `{self._strategy.__class__.__name__}` does not support skipping the gradient synchronization."
                 f" Remove `.no_backward_sync()` from your code or choose a different strategy.",
                 category=PossibleUserWarning,
             )
-            context = nullcontext()
-        else:
-            context = self._strategy._backward_sync_control.no_backward_sync(  # type: ignore[assignment]
-                module._forward_module
-            )
+            return nullcontext()
+        return self._strategy._backward_sync_control.no_backward_sync(module._forward_module)
 
-        with context:
-            yield
-
-    @contextmanager
-    def sharded_model(self) -> Generator:
+    def sharded_model(self) -> ContextManager:
```

Comment:
Why do these get converted? For consistency?
For the graph breaks, we only need the precision related ctx managers to changed since they apply in forward of the wrapper.

Reply:
These are not as important, but if somebody was doing a function that use them, they would have this problem too.